### PR TITLE
feat: first implementation

### PR DESCRIPTION
**Pull request by hardfist**
use module graph to purge tailwind css

### CODE REVIEW

**Review by colinaaa — APPROVED** [review]

File: src/index.mjs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                ...config.tailwindConfig.content,
+              ];
+              // iterate all css asset in entry and inject entry_modules into tailwind content
+              for (const file of entrypoint.getFiles()) {
+                const asset = compilation.getAsset(file);
+                if (file.endsWith(".css") && asset) {
+                  const content = asset.source.source();
+                  // transform .css which contains tailwind mixin
+                  // FIXME: add custom postcss config
+                  const transformed_css = await postcss([ tailwind({...config.tailwindConfig,content:all_contents})]).process(
```
Comment (chenjiahan) [inline src/index.mjs:40]: We do not need to init the plugin for every file:

```js
const tailwindPlugin = tailwind({...config.tailwindConfig,content:all_contents});

for (const file of entrypoint.getFiles()) {
```

File: src/index.mjs
Comment (chenjiahan) [inline src/index.mjs:1]: How about rewriting this plugin in TS and using Rslib to build it?

File: src/index.mjs (hunk 1 above)
Comment (hardfist) [reply src/index.mjs:40]: yes, I also need move postcss init upward

File: src/index.mjs
Comment (hardfist) [reply src/index.mjs:1]: would give it a try

**Pull request merged**